python/ray/_private/structured_logging.py

```diff
@@ -0,0 +1,42 @@
+import logging
+import ray
+import json
+
+from ray._private.ray_constants import (
+    LOG_RAY_CORE_CONTEXT,
+    LOGGER_RAY_ATTRS,
+    LOGRECORD_STANDARD_ATTRS,
+)
+
+
+class CoreContextFilter(logging.Filter):
+    def filter(self, record):
+        runtime_context = ray.get_runtime_context()
+        ray_core_context = {
+            "job_id": runtime_context.get_job_id(),
+            "worker_id": runtime_context.get_worker_id(),
+            "node_id": runtime_context.get_node_id(),
+        }
+        if runtime_context.worker.mode == ray.WORKER_MODE:
+            ray_core_context["actor_id"] = runtime_context.get_actor_id()
+            ray_core_context["task_id"] = runtime_context.get_task_id()
+        setattr(record, LOG_RAY_CORE_CONTEXT, ray_core_context)
+        return True
+
+
+class JSONFormatter(logging.Formatter):
+    def format(self, record):
+        super().format(record)
+        record_format = {
+            "ts": self.formatTime(record, self.datefmt),
+            "level": record.levelname,
+            "msg": record.getMessage(),
+            "filename": record.filename,
+            "lineno": record.lineno,
```

Comment:
Let's also record exec if exists

Reply:
What is `exec`? I don't see `exec` in either LogRecord's document and source code.

https://docs.python.org/3/library/logging.html#logrecord-objects

Reply:
Yea, exceptions